contracts/WildcardSteward_v2.sol

```diff
@@ -466,17 +530,110 @@ contract WildcardSteward_v2 is Initializable {
         address currentOwner = assetToken.ownerOf(tokenId);
         address tokenPatron = currentPatron[tokenId];
 
-        if (state[tokenId] == StewardState.Owned) {
-            uint256 totalToPayBack = price[tokenId];
-            // NOTE: pay back the deposit only if it is the only token the patron owns.
-            if (
-                totalPatronOwnedTokenCost[tokenPatron] ==
-                price[tokenId].mul(patronageNumerator[tokenId])
-            ) {
-                totalToPayBack = totalToPayBack.add(deposit[tokenPatron]);
-                deposit[tokenPatron] = 0;
-            }
+        _buy(tokenId, tokenPatron, wildcardsPercentage);
+        _handleBuy(
+            tokenId,
+            wildcardsPercentage,
+            remainingValueForDeposit,
+            currentOwner,
+            tokenPatron,
+            _newPrice
+        );
+    }
+
+    // Check possible issues with collectPatronage(tokenId) [probably leave out since no patronage to collect on forecolsed token!?]
+    function buyAuction(
+        uint256 tokenId,
+        uint256 _newPrice,
+        uint256 wildcardsPercentage
+    )
+        public
+        payable
+        collectPatronage(tokenId)
+        collectPatronageAddress(msg.sender)
+        priceGreaterThanZero(_newPrice)
+        validWildcardsPercentage(wildcardsPercentage)
+    {
+        require(
+            state[tokenId] == StewardState.Foreclosed,
+            "Can only buy foreclosed tokens useing this function"
+        );
+        // Calculate auction price for the token and replace price[tokenId] below
+        uint256 auctionTokenPrice = _auctionPrice(tokenId);
+
+        // Note: We need to calculate the requisite value on our front end using same function
+        uint256 remainingValueForDeposit = msg.value.sub(auctionTokenPrice);
+
+        address currentOwner = assetToken.ownerOf(tokenId);
+        address tokenPatron = currentPatron[tokenId];
+
+        _buyAuction(tokenId);
+        _handleBuy(
+            tokenId,
+            wildcardsPercentage,
+            remainingValueForDeposit,
+            currentOwner,
+            tokenPatron,
+            _newPrice
+        );
+    }
+
+    function _handleBuy(
+        uint256 tokenId,
+        uint256 wildcardsPercentage,
+        uint256 remainingValueForDeposit,
+        address currentOwner,
+        address tokenPatron,
+        uint256 _newPrice
+    ) internal {
+        wildcardsPercentages[tokenId] = wildcardsPercentage; //Set the percentage wildcards will recieve on next sale
+        deposit[msg.sender] = deposit[msg.sender].add(remainingValueForDeposit);
+        transferAssetTokenTo(
+            tokenId,
+            currentOwner,
+            tokenPatron,
+            msg.sender,
+            _newPrice
+        );
+        emit Buy(tokenId, msg.sender, _newPrice);
+    }
+
+    // NB to check that the msg.value is greater tgat
+    function _buyAuction(uint256 tokenId) internal {
```

Comment:
I somehow think it is easier to read the code if it is all sequential in the buyAuction function. I don't think it is much of readability gain putting it in its own funcion like this, and it isn't re-used anywhere either.

Reply:
Ok, thought about it more, it isn't that bad having this in its own function, but the name is wrong; it should say something like `distributeTokenSaleEarnings` or something like that. Also, I think it is maybe a bit stingy to only give the artist 1% on the initial sale... It is going to complicate our logic to do anything else, so I don't think it is a good idea; but in the future we should definitely plan to give a big portion to the artist on initial sale (and then a small percentage thereafter).

Reply:
`distributeAuctionProceeds` :)